apps/src/templates/teacherDashboard/RosterDialog.jsx

```diff
@@ -151,23 +151,37 @@ class RosterDialog extends React.Component {
     classrooms: PropTypes.arrayOf(classroomShape),
     loadError: loadErrorShape,
     rosterProvider: PropTypes.oneOf(Object.keys(OAuthSectionTypes)),
+    userId: PropTypes.number,
   };
 
   state = {selectedId: null};
 
-  importClassroom = () => {
+  importClassroom = redirect => {
     this.recordSectionSetupExitEvent(COMPLETED_EVENT);
     const classrooms = this.props.classrooms;
     const selectedName =
       classrooms &&
       classrooms.find(classroom => {
         return classroom.id === this.state.selectedId;
       }).name;
-
-    this.props.handleImport(this.state.selectedId, selectedName);
+    this.props.handleImport(this.state.selectedId, selectedName, redirect);
     this.setState({selectedId: null});
   };
 
+  // create new function for redirect to NewEditPage
+  redirectToEditSectionPage = sectionId => {
+    const redirectUrl = '/sections/' + sectionId + '/edit';
+    console.log(redirectUrl);
+    window.location.href = redirectUrl;
+  };
+
+  // Need to update with section ID
+  handleRedirect = () => {
+    const redirect = true;
+    this.importClassroom(redirect);
+    this.redirectToEditSectionPage(/*sectionId*/);
```

Comment:
Ahh okay I see how this would be tricky. The oauth section will have a very freshly created id... I have a feeling we'll have to query it from the api- I'll look into it!